File: backend/app/models/food.py
from sqlalchemy import Column, Integer, String, Float
from sqlalchemy.dialects.postgresql import TSVECTOR
from app.core.database import Base

class Food(Base):
    __tablename__ = "foods"

    id = Column(Integer, primary_key=True)
    name = Column(String, index=True)

    calories = Column(Float)
    protein = Column(Float)
    carbs = Column(Float)
    fat = Column(Float)

    search_vector = Column(TSVECTOR)

File: frontend/types/food.ts
export interface Food {
  id: number;
  name: string;
  calories: number;
  protein: number;
  carbs: number;
  fat: number;
}

export interface MealSummary {
  total_calories: number;
  total_protein: number;
  total_carbs: number;
  total_fat: number;
}

export interface Meal {
  meal_id: number;
}